### PR TITLE
Pub 696 - filter for alphabet search

### PR DESCRIPTION
**Pull request by kurt-uk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-696

### Change description ###
added remaining tests that were missing from the sudden merge of 696 to make demo time, changed some logic in controller that stopped get req taking off filter names properly

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/service/filterService.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import {Court} from '../models/court';
 
+const filterNames = ['Jurisdiction', 'Region'];
+
 export class FilterService {
-  public getFilterValueOptions(filterName: string, list: Array<Court>): string[] {
+  private getFilterValueOptions(filterName: string, list: Array<Court>): string[] {
```
Comment (commitBlob) [inline src/main/service/filterService.ts:6]: by making this private it is not accessible in tests
My linter is screaming  Property 'getFilterValueOptions' is private and only accessible within class 'FilterService' when ac

File: src/test/unit/service/filterService.test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const rawData = fs.readFileSync(path.resolve(__dirname, '../mocks/courtAndHearings.json'), 'utf-8');
+const listData = JSON.parse(rawData);
+
+const crownCourt = 'Crown Court';
+const royalCourt = 'Royal Court';
+const jurisdiction = 'Jurisdiction';
+const filterOptions = {Jurisdiction: {'Crown Court': {checked: true}, Crown: {checked: false}}, Region: {Bedford: {checked: false}, Hull: {checked: true}}};
+
+describe('Filter Service', () => {
```
Comment (commitBlob) [inline src/test/unit/service/filterService.test.ts:15]: No tests for getFilterValueOptions?

File: src/main/service/filterService.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
     return [...new Set(list.map(court => court[filterName.toLowerCase()]))];
   }
 
-  public buildFilterValueOptions(filterNames: string[], list, selectedFilters: string[]): object {
+  public buildFilterValueOptions(list, selectedFilters: string[]): object {
```
Comment (commitBlob) [inline src/main/service/filterService.ts:10]: based on your comment 'if we can start being little more accurate with return types'
shouldn't we do the same for attributes passed in?

File: src/main/service/filterService.ts [hunk 4]
```diff
@@ -29,4 +31,20 @@ export class FilterService {
       return selectedFilters;
     }
   }
+
+  public handleKeys(filterOptions: object, filterValues: string[]) {
```
Comment (commitBlob) [inline src/main/service/filterService.ts:35]: if we can start being little more accurate with return types

File: src/main/service/filterService.ts (hunk 1 above)
Comment (kurt-uk) [reply src/main/service/filterService.ts:6]: i set it to private as it it not needed outside this class, it is only used within filter service, its unit tests are covered by publicly accessible tests

File: src/test/unit/service/filterService.test.ts (hunk 2 above)
Comment (kurt-uk) [reply src/test/unit/service/filterService.test.ts:15]: no its a private method? you dont test private methods directly you test them passively

**Review by commitBlob — DISMISSED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Review by commitBlob — APPROVED** [review]

**Pull request merged**